qa-tests-backend/src/main/groovy/util/CollectorUtil.groovy

```diff
@@ -12,6 +19,59 @@ class CollectorUtil {
     static final String ENABLED_VALUE = "ENABLED"
     static final String DISABLED_VALUE = "DISABLED"
 
+    static parseJsonToProtobuf(String json) {
+        def builder = sensor.Collector.CollectorConfig.getMethod("newBuilder").invoke(null)
+        JsonFormat.parser().merge(json, builder)
+        return builder.build()
+    }
+
+    static introspectionQuery(String collectorIP, String endpoint) {
+        def uri = "http://${collectorIP}:8080${endpoint}"
+        def connection = new URL(uri).openConnection() as HttpURLConnection
+    
+        try {
+            connection.requestMethod = "GET"
+            connection.connect()
+    
+            if (connection.responseCode != HttpURLConnection.HTTP_OK) {
+                throw new RuntimeException("IntrospectionQuery failed with ${connection.responseMessage}")
+            }
+            def jsonResponse = connection.inputStream.text
+            return parseJsonToProtobuf(jsonResponse)
+        } catch (Exception e) {
+            throw new RuntimeException("Error making request: ${e.message}", e)
+        } finally {
+            connection.disconnect()
+        }
+    }
+
+    static waitForConfigToHaveState(String state, int timeoutSeconds = 90) {
+        int intervalSeconds = 1
+        int waitTime
+        def startTime = System.currentTimeMillis()
+        for (waitTime = 0; waitTime <= timeoutSeconds / intervalSeconds; waitTime++) {
+            def config = introspectionQuery("127.0.0.1", "/state/runtime-config")
```

Comment:
This won't work in CI. Need to get the actual IP address. Also need to enable collector introspection and port-forward collector.